plugins/modules/cloud_provider_account.py

```diff
@@ -0,0 +1,201 @@
+            for a in p['accounts']:
+                if a['id'] == module.params['id']:
+                    account = a
+                    break
+            break
+    
+    # Delete account if needed
+    if account and module.params['provider'] == 'present':
+        if module.check_mode:
+            return_changed(module)
+        delete_body = {
+            'id': module.params['id'],
+        }
+        path = '/cloud_access_providers/'+module.params['provider']+'/accounts'
+        delete = client.delete(path, data=delete_body)
+        if delete.status_code != 204:
+            module.fail_json(msg="Failed to delete provider account")
+        result['changed'] = True
+        module.exit_json(**result)
+    
+    # Create account if missing
+    if not account:
+        if module.check_mode:
+            return_changed(module)
+        new_account = [
+            {
+                'id': module.params['id'],
+                'nickname': module.params['nickname'],
+            }
+        ]
+        path = '/cloud_access_providers/'+module.params['provider']+'/accounts'
+        create = client.post(path, data=new_account)
+        if create.status_code != 204:
+            module.fail_json(msg="Failed to create provider account")
+        result['changed'] = True
+    else:
+        # Update existing account if nickname is different
+        if module.params['nickname'] and module.params['nickname'] != account['nickname']:
+            if module.check_mode:
+                return_changed(module)
+            updated_account = {'nickname': module.params['nickname']}
+            path = '/cloud_access_providers/'+module.params['provider']+'/accounts/'+account['id']
+            update = client.put(path, data=updated_account)
+            if update.status_code != 204:
+                module.fail_json(msg="Failed to update provider account")
+            result['changed'] = True
+    
+    # Submit an account verification if provided and account is not already verified
+    if module.params['verification_identity'] and 'verified' in account and not account['verified']:
+        if module.check_mode:
+            return_changed(module)
+        # The API expects identity and signature to be base64 encoded
+        # Because base64.b64encode returns values as bytes, we decode it back to a string
+        verification = {
+            'identity': module.params['verification_identity'],
+            'signature': module.params['verification_signature'],
+        }
+        path = '/cloud_access_providers/'+module.params['provider']+'/accounts/'+module.params['id']+'/verification'
+        verify = client.put(path, data=verification)
+        if verify.status_code != 204:
+                module.fail_json(msg="Failed to verify provider account")
+        result['changed'] = True
+
+    module.exit_json(**result)
+
+
+def main():
+    run_module()
+
+
+if __name__ == '__main__':
+    main()
```

Comment:
Do we care about newlines at end of files? I have my editor set to add them by default but I've never really encountered a tool that fails without it present.

Reply:
Probably should for consistency's sake.